source/MaterialXRuntime/RtNode.h

```diff
@@ -60,8 +63,13 @@ class RtNode : public RtTypedSchema
     /// Return an iterator traversing all output attributes
     /// on this node. Shorthand for getPrim().getOutputs().
     RtAttrIterator getOutputs() const;
+
+    /// Returns a vector of public nodegraph metadata
+    const RtTokenVec& getPublicMetadata() const override;
```

Comment:
I'm still on the fence about the method name. Maybe `getPublicMetadataNames` is better since it's just a list of the names of the metadata and not the actual metadata. We can always update later though (before the API is released).